src/main/java/com/minecolonies/coremod/client/gui/WindowBuildBuilding.java

```diff
@@ -0,0 +1,353 @@
+package com.minecolonies.coremod.client.gui;
+
+import com.minecolonies.api.crafting.ItemStorage;
+import com.minecolonies.api.util.BlockUtils;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.api.util.LanguageHandler;
+import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.blockout.Pane;
+import com.minecolonies.blockout.controls.Button;
+import com.minecolonies.blockout.controls.ButtonHandler;
+import com.minecolonies.blockout.controls.ItemIcon;
+import com.minecolonies.blockout.controls.Label;
+import com.minecolonies.blockout.views.DropDownList;
+import com.minecolonies.blockout.views.ScrollingList;
+import com.minecolonies.blockout.views.Window;
+import com.minecolonies.coremod.MineColonies;
+import com.minecolonies.coremod.blocks.ModBlocks;
+import com.minecolonies.coremod.colony.ColonyView;
+import com.minecolonies.coremod.colony.StructureName;
+import com.minecolonies.coremod.colony.Structures;
+import com.minecolonies.coremod.colony.buildings.views.AbstractBuildingView;
+import com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIStructure;
+import com.minecolonies.coremod.network.messages.BuildRequestMessage;
+import com.minecolonies.coremod.network.messages.BuildingSetStyleMessage;
+import com.minecolonies.coremod.util.StructureWrapper;
+import com.minecolonies.structures.helpers.Settings;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockBed;
+import net.minecraft.block.BlockDoor;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Mirror;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.gen.structure.template.Template;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+import static com.minecolonies.api.util.constant.TranslationConstants.*;
+import static com.minecolonies.api.util.constant.WindowConstants.*;
+
+/**
+ * Window for selecting the style and confirming the resources.
+ */
+public class WindowBuildBuilding extends AbstractWindowSkeleton implements ButtonHandler
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') "ButtonHandler" is implemented by a super class; there is no need to implement it here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1939)